src/vm/threads.cpp

```diff
@@ -2144,6 +2144,30 @@ HANDLE Thread::CreateUtilityThread(Thread::StackSizeBucket stackSizeBucket, LPTH
 }
 
 
+// Represent the value of OVERRIDE_DEFAULT_STACK_SIZE_4KPAGES as passed in the property bag to the host during construction
+static uint32_t s_overrideDefaultStackSize = 0;
+
+void OverrideDefaultStackSize(LPCWSTR valueStr)
+{
+    if (valueStr)
+    {
+        LPWSTR end;
+        errno = 0;
+        unsigned long value = wcstoul(valueStr, &end, 10);
+
+        SIZE_T max4KPages = SIZE_T(1) << (31 - 12);
+
+        if ((errno == ERANGE) || (valueStr == end) || (end == nullptr) || (end[0] != 0) || (value >= max4KPages))
+        {
+            ThrowMessage("OVERRIDE_DEFAULT_STACK_SIZE_4KPAGES value parse error");
+        }
+        else
+        {
+            FastInterlockExchange((int32_t*)&s_overrideDefaultStackSize, (uint32_t) value * 4096);
```

Comment:
Why do we need interlocked operation here? Do you expect multiple threads to race for setting the value? It seems that it is called only on the initial thread.

Reply:
It will only be used at startup.  I was thinking it needed a barrier on ARM systems to make sure the stack size was visible to all threads, but I am certain that thread create will already add a barrier.  I'll remove it.